### PR TITLE
 Moves end round info to its own panel, second try

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Reopening of https://github.com/tgstation/TerraGov-Marine-Corps/pull/12151 without the strange commits.

You know how when you play regular SS13 and you finish a round a panel pops out in the middle of your screen with the round info? This is that but for TGMC.
I mostly used code from TG for this, but there's a few minor areas that received minor rewrites.
It should display all the info and stats that are normally displayed at the end of a round, I've verified this locally but it needs a TM to be entirely sure.

Pictures soon.
~~A week later apparently~~
endround

## Why It's Good For The Game

Have you ever had a round end, the stats and results get pushed to chat and then it scrolls so fast you have to scroll up for five minutes to find out who won, fighting against the tide of new chat messages? This should put an end to that for good, the panel is placed in the middle of the screen and there's a handy button for reopening it.

To my knowledge, CM and TGMC are the only codebases that do not push a notification to the screen on round end, and I intend to change this. ~~I also need this for future features I'm working on~~

## Changelog
:cl:
qol: Round end info is now displayed in its own panel instead of in chat
/:cl:

### CODE REVIEW

**Comment by SilencedMP5A5** [comment]
when do we get more detailed round statistics

**Comment by RipGrayson** [comment]
> when do we get more detailed round statistics

Whenever they're added, currently this just pushes any round stats you would normally see in chat to its own panel.

**Comment by Joe13413** [comment]
powercreep pr

**Comment by Al-1ce** [comment]
Allowing a preference to toggle this between UI and the chat is a necessity. If that isn't plausible to add, I really do not like this.

File: code/datums/gamemodes/_game_mode.dm
Comment (TiviPlus) [inline code/datums/gamemodes/_game_mode.dm:754]: Autodoc

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm

**Comment by Ragecrabbo** [comment]
Show roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report againShow roundend report again

File: code/datums/gamemodes/_game_mode.dm
Comment (Lumipharon) [inline code/datums/gamemodes/_game_mode.dm:207]:
```suggestion
	display_report()
```

File: code/datums/gamemodes/_game_mode.dm
Comment (Lumipharon) [inline code/datums/gamemodes/_game_mode.dm:710]:
```suggestion
/datum/game_mode/proc/display_report()
```
popcount isn't used unless I'm entirely blind

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by XSlayer300** [comment]
Make sure this ONLY appears if you've joined the game proper (as a marine/ERT or a xeno) and not if you just observed the round or somehow killed yourself in vahalla.
image

**Comment by RipGrayson** [comment]
> Make sure this ONLY appears if you've joined the game proper (as a marine/ERT or a xeno) and not if you just observed the round or somehow killed yourself in vahalla. image

That sort of thing will probably be implemented later, I would have to touch the jobs system and massively complicate what's already written.

**Comment by Lumipharon** [comment]
It also uses the hvh end game wording in hvx, referring to the brave men and women of the tgmc and som

**Comment by XElectricX** [comment]
Finally don't have to scroll up to see the round end stats, though having it as a preference sounds fair. Also what exactly is the button sprite supposed to be? Looks very bad.

**Comment by RipGrayson** [comment]
> Finally don't have to scroll up to see the round end stats, though having it as a preference sounds fair. 

Probably something that will happen later, you'll need two procs for each displayed statistic and to modify both the preferences panel and the logic involved in displaying end round stats. It's an extra complication that can wait until after the PR itself has been merged.

>Also what exactly is the button sprite supposed to be? Looks very bad.

Feel free to make a new one, it's a codersprite and I have no ability/inspiration to make a new one.

**Comment by TiviPlus** [comment]
There will be no preference

**Comment by Al-1ce** [comment]
> There will be no preference

Cringe.

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**
